main.py

```diff
@@ -263,25 +254,42 @@ def _kill_tb_child():
     atexit.register(_kill_tb_child)
 
 
-def get_best_checkpoint_path(run_dir, phase, task_name=None):
+def get_best_checkpoint_path(args, phase, task_name=None):
     """ Look in run_dir for model checkpoint to load.
     Hierarchy is
-        1) best task-specific checkpoint for target_train, used when evaluating
+        1) user-specified target task checkpoint
```

Comment:
Write out the full logic, including which of these cases are phase-specific. If the logic looks too hairy, try to simplify the logic?

Reply:
As-is, this would make me suspect that if I supplied target and eval checkpoints, we'd load the target one regardless of the phase.

Reply:
Also, see the comment in the other PR: Basically, when we're starting a phase, we should always check for checkpoints from the previous phase. That way, we can rerun evaluation for a job by simply setting do_target_train = 0 and do_full_eval = 1. Pretty common, for example, in cases where we have a new eval or analysis file to use.